troop/views.py

```diff
@@ -0,0 +1,92 @@
+from django.contrib.auth.mixins import AccessMixin
+from django.urls import reverse
+from django.views import generic
+from django.contrib import messages
+from django.utils.translation import gettext_lazy as _
+from django.db.models import Count
+
+from .models import Participant
+from .forms import CreateParticipantForm
+
+
+class OnlyTroopManagerMixin(AccessMixin):
+    """Verify that the current user is allowed to manage the troop."""
+
+    def dispatch(self, request, *args, **kwargs):
+        if not request.user.is_authenticated:
+            return self.handle_no_permission()
+
+        troop_number = kwargs.get("troop")
+        if not troop_number:
+            return self.handle_no_permission()
+
+        troop = request.user.troops.filter(number=troop_number).first()
```

Comment:
What if a user manages two troops?

Reply:
That's not a problem.

Here we are searching **among the troops attached of the user** the one with the given `troop_number`.

![image](https://user-images.githubusercontent.com/3864879/75385113-76a99d00-58df-11ea-9f2d-5b5098e0e3f4.png)

Reply:
I see, alright :)